test/unit/services/NamedQueryService.test.js

```diff
@@ -0,0 +1,142 @@
+describe('The Named Query Service', function () {
+  before(function (done) {
+    done()
+  })
+
+  it("Build a valid query with default values", function (done) {
+
+    const namedQueryConfig = {
+      collectionName: "record",
+      brandIdFieldPath: "metaMetadata.brandId",
+      resultObjectMapping: {
+
+      },
+      mongoQuery: {
+        "daysDatePath": null,
+        "isoDatePath": null,
+        "stringPath": null,
+        "numberPath": null
+      },
+      queryParams: {
+
+        daysDateParam: {
+
+          type: "date",
+          path: "daysDatePath",
+          queryType: "<=",
+          format: "days",
+          whenUndefined: "defaultValue",
+          defaultValue: "-1"
+        },
+        isoDateParam: {
+          type: "date",
+          queryType: "=>",
+          path: "isoDatePath",
+          format: "isoDate",
+          whenUndefined: "defaultValue",
+          defaultValue: "2024-12-17T00:00:00.000Z"
+        },
+        stringParam: {
+          type: "string",
+          path: "stringPath",
+          queryType: 'contains',
+          whenUndefined: "defaultValue",
+          defaultValue: "defaultString"
+        },
+        //TODO: number doesn't support queryType of default value like the others
+        numberParam: {
+          type: "number",
+          path: "numberPath"
+        }
+
+      }
+
+    };
+
+    const numberParamValue = 1;
+    let paramMap = {numberParam: numberParamValue};
+
+    NamedQueryService.setParamsInQuery(namedQueryConfig.mongoQuery, namedQueryConfig.queryParams, paramMap);
+    expect(namedQueryConfig.mongoQuery.numberPath).to.equal(numberParamValue);
+    expect(namedQueryConfig.mongoQuery.isoDatePath).to.deep.equal({ "=>": '2024-12-17T00:00:00.000Z'});
+    expect(namedQueryConfig.mongoQuery.stringPath).to.deep.equal({contains: 'defaultString'});
+
+    const daysDateString = namedQueryConfig.mongoQuery.daysDatePath["<="]
+    expect(daysDateString).to.be.a('string');
+    let yesterdayIsoString = moment().subtract(1, 'days').format('YYYY-MM-DD');
+    expect(daysDateString).to.contain(yesterdayIsoString);
+    
+    done()
+  })
+
+
+  it("Build a valid query with passed values", function (done) {
+
+    const namedQueryConfig = {
+      collectionName: "record",
+      brandIdFieldPath: "metaMetadata.brandId",
+      resultObjectMapping: {
+
+      },
+      mongoQuery: {
+        "daysDatePath": null,
+        "isoDatePath": null,
+        "stringPath": null,
+        "numberPath": null
+      },
+      queryParams: {
+
+        daysDateParam: {
+
+          type: "date",
+          path: "daysDatePath",
+          queryType: "<=",
+          format: "days",
+          whenUndefined: "defaultValue",
+          defaultValue: "-1"
+        },
+        isoDateParam: {
+          type: "date",
+          queryType: "=>",
+          path: "isoDatePath",
+          format: "isoDate",
```

Comment:
Is the format name `isoDate` or `ISODate`? The enum uses `ISODate`.

Perhaps the date branch needs to check the format is one of the enums, and fail if it isn't?

Reply:
Good pickup. I've changed it to ISODate. the "format" option was added later to allow the days feature, for that reason I think it should default to ISODate when not provided. I've added to the tests a check for that.